src/client/websocket/WebSocketManager.js

```diff
@@ -1,34 +1,74 @@
-const EventEmitter = require('events');
-const { Events, Status } = require('../../util/Constants');
-const WebSocketConnection = require('./WebSocketConnection');
+const WebSocketShard = require('./WebSocketShard');
+const { Events, Status, WSEvents } = require('../../util/Constants');
+const PacketHandlers = require('./handlers');
+
+const BeforeReadyWhitelist = [
+  WSEvents.READY,
+  WSEvents.RESUMED,
+  WSEvents.GUILD_CREATE,
+  WSEvents.GUILD_DELETE,
+  WSEvents.GUILD_MEMBERS_CHUNK,
+  WSEvents.GUILD_MEMBER_ADD,
+  WSEvents.GUILD_MEMBER_REMOVE,
+];
 
 /**
  * WebSocket Manager of the client.
  * @private
  */
-class WebSocketManager extends EventEmitter {
+class WebSocketManager {
   constructor(client) {
-    super();
     /**
      * The client that instantiated this WebSocketManager
      * @type {Client}
      */
     this.client = client;
 
     /**
-     * The WebSocket connection of this manager
-     * @type {?WebSocketConnection}
+     * The gateway this WebSocketManager uses.
+     * @type {string}
+     */
+    this.gateway = undefined;
+
+    /**
+     * An array of shards spawned by this WebSocketManager.
+     * @type {WebSockethard[]}
```

Comment:
should be `WebSocketShard`, also would there be any use in mapping the shard ID to the shard instead of using an array?